src/3d/qgs3dmapscene.cpp

```diff
@@ -108,6 +109,18 @@ Qgs3DMapScene::Qgs3DMapScene( const Qgs3DMapSettings &map, QgsAbstract3DEngine *
   connect( &map, &Qgs3DMapSettings::fieldOfViewChanged, this, &Qgs3DMapScene::updateCameraLens );
   connect( &map, &Qgs3DMapSettings::renderersChanged, this, &Qgs3DMapScene::onRenderersChanged );
 
+  connect( Qgs3DModelCache::instance(), &Qgs3DModelCache::remote3DModelFetched, this, [ = ]
+  {
+    const QList<QgsMapLayer *> pointVectorLayers = mPointVectorLayers;
+    for ( QgsMapLayer *layer : pointVectorLayers )
+    {
+      // remove old entity - if any
+      removeLayerEntity( layer );
+      // add new entity - if any 3D renderer
+      addLayerEntity( layer );
+    }
+  } );
```

Comment:
Would it be possible to put handling of the "model fetched" signal to qgspoint3dsymbol_p.cpp? I would like to avoid too many implementation details leaking into 3D map scene code...

Also, it looks like it would rebuild all point layers - even those that are not using that model at all...

Reply:
@wonder-sk , I'm open to suggestions as to where you think we can plug in the model fetched signal in there. I couldn't see a way, but that definitively doesn't mean it can't be done :wink:

Reply:
(actually on that, if we leave it in 3D map scene, we can narrow down the list of layers to be rebuild by only adding point layers with models as 3D symbols.

Reply:
I would probably create a new class, e.g. QgsModelPoint3DSymbolEntity derived from QEntity. It could listen to the signal and refresh its internal QSceneLoader if URI matches. Then QgsModelPoint3DSymbolHandler::addSceneEntities() would just create instances of that class...